### ISSUE TITLE
In Flow 0.13.1, type of `+foo` isn't number

### ISSUE THREAD

**Issue by danvk** [facebook/flow#588]
This snippet type checks in Flow 0.12.0, but not 0.13.1:

```javascript
/** @flow */

function foo(): number {
  var num = '123',
      v = +num;
  // Test for wraparound & roundoff
  if (v < 0 || 1 + v == v) {
    throw new RangeError(`Number out of precise floating point range: ${num}`);
  }
  return v;
}
```

The error in Flow 0.13.1 indicates that the type of `v` is `string`. When I put it under my cursor in my editor and run `:FlowType`, I get `any`. The type of `+x` should always be `number`.

```
$ cat toy.js | flow check-contents

-:7:16,20: +
Error:
-:7:16,16: number
This type is incompatible with
-:7:16,20: string

Found 1 error
```

**Comment by avikchaudhuri** [comment]
Related to #587.

**Comment by mroch** [comment]
I think this was exposed by #578 (to which #587 is related), but I don't think it's new. Check this out: https://github.com/facebook/flow/blob/b4d380dab715cf511a5c9db29b32a5aee4386bcc/src/typing/type_inference_js.ml#L3396

**Comment by mroch** [comment]
Fixed by f64bc2ea44e832422f645f5575da7485c29cd708

**Comment by danvk** [comment]
Thanks for the quick fix!
